hist/hist/inc/TScatter.h

```diff
@@ -0,0 +1,57 @@
+// @(#)root/hist:$Id$
+// Author: Olivier Couet   18/05/2022
+
+/*************************************************************************
+ * Copyright (C) 1995-2000, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_TScatter
+#define ROOT_TScatter
+
+
+//////////////////////////////////////////////////////////////////////////
+//                                                                      //
+// TScatter                                                             //
+//                                                                      //
+// A scatter plot able to draw four variables on a single plot          //
+//                                                                      //
+//////////////////////////////////////////////////////////////////////////
+
+#include "TGraph.h"
+class TH1F;
+class TScatter : public TGraph {
+
+protected:
+   Double_t  *fColor{nullptr};   ///< [fNpoints] array of colors
+   Double_t  *fSize{nullptr};    ///< [fNpoints] array of marker sizes
```

Comment:
I would use `std::vector<Double_t>` instead of plain C arrays.

Reply:
I matched waht is done in TGraph.h for fX and fY. That would be painful to have one part of the vectors defined as normal vectors and the other part as std::vector. I think we already add this discussion when i created the tsctter branch (before creating the PR)

Reply:
New code not need to be exactly match with old. Usage of vectors is more safe.
But this is not important. You can keep it as is.

Reply:
Yes I would prefer to do a global change in all TGraphxxx .